yt/frontends/cholla/fields.py

```diff
@@ -0,0 +1,134 @@
+
+
+    def setup_fluid_fields(self):
+        # Here we do anything that might need info about the dataset.
+        # You can use self.alias, self.add_output_field (for on-disk fields)
+        # and self.add_field (for derived fields).
+
+        unit_system = self.ds.unit_system
+
+        # Add velocity fields
+        for comp in "xyz":
+              self.add_field(
+                  ("gas", f"velocity_{comp}"),
+                  sampling_type="cell",
+                  function=velocity_field(comp),
+                  units=unit_system["velocity"],
+              )
+
+        def kinetic_energy_density(data):
+            return (
+                     0.5*data["cholla", "density"]
+                     * (data["gas", "velocity_x"] * data["gas", "velocity_x"]
+                      + data["gas", "velocity_y"] * data["gas", "velocity_y"]
+                      + data["gas", "velocity_z"] * data["gas", "velocity_z"]
+                     )
+            )
+
+        # Add pressure field
+        if ("cholla", "GasEnergy") in self.field_list:
+           self.add_output_field(
+               ("cholla", "GasEnergy"), sampling_type="cell", units=pres_units
+           )
+           self.alias(
+               ("gas", "thermal_energy"),
+               ("cholla", "GasEnergy"),
+               units=unit_system["pressure"],
+           )
+
+           def _pressure(field, data):
+               return (
+                  (data.ds.gamma - 1.0)
+                  * data["cholla", "GasEnergy"]
+               )
+        else:
+          def _pressure(field, data):
+              return (
+                  (data.ds.gamma - 1.0)
+                  * (data["cholla", "Energy"] - 
+                     kinetic_energy_density(data)
+                    )
+                  )
+
+        self.add_field(
+            ("gas", "pressure"),
+            sampling_type="cell",
+            function=_pressure,
+            units=unit_system["pressure"],
+        )
+        
+        def _specific_total_energy(field, data):
+            return data["cholla", "Energy"] / data["cholla", "density"]
+
+        self.add_field(
+            ("gas", "specific_total_energy"),
+            sampling_type="cell",
+            function=_specific_total_energy,
+            units=unit_system["specific_energy"],
+        )
+
+
+        # Add temperature field
+        def _temperature(field, data):
+            return (
+                data.ds.mu
+                * data["gas", "pressure"]
+                / data["gas", "density"]
+                * mh
+                / kboltz
+            )
+
+        self.add_field(
+            ("gas", "temperature"),
+            sampling_type="cell",
+            function=_temperature,
+            units=unit_system["temperature"],
+        )
+        
+
+    def setup_particle_fields(self, ptype):
+        super().setup_particle_fields(ptype)
+        # This will get called for every particle type.
```

Comment:
Same as above. No need to override a method if we don't do anything more than the parent class
```suggestion
```